src/Moryx.ControlSystem/Cells/ReadyToWork.cs

```diff
@@ -52,20 +52,16 @@ internal ReadyToWork(Session currentSession)
         /// </summary>
         /// <param name="activity">The activity.</param>
         public ActivityStart StartActivity(IActivity activity)
-        {
-            // Update process before next stage
-            Process = activity.Process;
-            return new ActivityStart(this, activity);
+        {           
+            return new ActivityStart(this, activity) { Process = activity.Process };
         }
 
         /// <summary>
         /// Creates the SequenceCompleted message
         /// </summary>
         public SequenceCompleted CompleteSequence(IProcess process, bool processActive, params long[] nextCells)
         {
-            // Update process before next stage
-            Process = process;
```

Comment:
The only thing i don't get is why can this be removed here and in the `StartActivity` Method?

Reply:
It is not removed, it is in the object initializer now